sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/customlogic/CustomDs2to1Transformer.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2022 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.smartdatalake.workflow.action.spark.customlogic
+
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.workflow.dataobject.DataObject
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+
+import scala.reflect.runtime.universe.{Type, TypeTag, typeOf}
+
+/**
+ * Interface to define a custom Spark-Dataset transformation (1:1)
+ * When you implement this interface, you need to provide two case classes: One for your input Dataset
+ * and one for your output Dataset.
+ */
+trait CustomDs2to1Transformer[In1 <: Product, In2 <: Product, Out <: Product] extends Serializable {
+
+  /**
+   * Function to be implemented to define the transformation between an input and output DataFrame (1:1)
+   *
+   * @param session      Spark Session
+   * @param options      Options specified in the configuration for this transformation
+   * @param inputDS      Input Dataset
+   * @param dataObjectId Id of DataObject of SubFeed
+   * @return Transformed DataFrame
+   */
+
+  def transform(session: SparkSession, options: Map[String, String], ds1: Dataset[In1], ds2: Dataset[In2]): Dataset[Out]
+
+  def getTypeTag: Type = throw new IllegalArgumentException("When using option \"parameterResolution\" -> \"dataObjectId\"," +
```

Comment:
I'm curious if you see another way of doing this... I did not find any other way to get the the type tag of the User's Class other than forcing the user to pass it onto me